web/tests/pill_typeahead.test.js

```diff
@@ -160,31 +160,31 @@ run_test("set_up", ({mock_template, override}) => {
             if (opts.stream) {
                 // Test stream highlighter_html for widgets that allow stream pills.
                 assert.equal(
-                    config.highlighter_html.call(fake_stream_this, denmark),
+                    config.highlighter_html.call(fake_stream_this, denmark, fake_stream_this.query),
```

Comment:
If `highlighter_html` doesn’t use `this`, then we don’t need to pass one with `.call()`.

```suggestion
                    config.highlighter_html(denmark, fake_stream_this.query),
```

(Same for all the others.)

Reply:
Unfortunately for `composebox_typeahead`, it still uses `this` for some things. I didn't bother cleaning that up in this PR because it was a bit more involved, though it would be nice to do that at some point. Actually I'll just do that now and add a couple more commits in here, why not.

Reply:
Removed all uses of `.call` :)